src/ng/sanitizeUri.js

```diff
@@ -72,11 +72,11 @@ function $$SanitizeUriProvider() {
     return function sanitizeUri(uri, isMediaUrl) {
       // if (!uri) return uri;
       var regex = isMediaUrl ? imgSrcSanitizationWhitelist : aHrefSanitizationWhitelist;
-      var normalizedVal = urlResolve(uri && uri.trim()).href;
+      var normalizedVal = urlResolve(uri && uri.toString().trim()).href;
       if (normalizedVal !== '' && !normalizedVal.match(regex)) {
         return 'unsafe:' + normalizedVal;
       }
-      return uri;
+      return uri.toString();
```

Comment:
For the record, this will allow more types than just `number`. (This might be a good thing, but it might also lead to unexpected results, instead of a fastfailure.)

If you want to explicitly only allow numbers and strings, you can have an explicit check (as we do in [$anchorScroll][1] for example).

[1]: https://github.com/angular/angular.js/blob/ebeb1c94917eaf53358ae0892232e6a7bb987ce2/src/ng/anchorScroll.js#L242

Reply:
yes, I was also thinking about this. I've added this mainly because it's weird to sanitize a string value but return the original value. For ngHref and other interpolation it's actually not necessary, as the interpolation itself does the string conversion. So maybe we should move the toString to where the value is passed into the sanitize fn?

Reply:
Sounds reasonable.

Reply:
@petebacondarwin what do you think about this? Should we stringify here or somewhere earlier in the chain? I have a hard time pinpointing where exactly this behavior changed. I guess it happened somewhere here: https://github.com/angular/angular.js/pull/16378

Reply:
Sounds good to me! I've updated the implementation and added a new test for custom toString(), too for good measure.